api/src/main/java/org/pmiops/workbench/cohortreview/AnnotationQueryBuilder.java

```diff
@@ -0,0 +1,219 @@
+      + "ON a%d.participant_id = pcs.participant_id "
+      + "AND a%d.cohort_annotation_definition_id = %d"
+      + "AND a%d.cohort_review_id = %d";
+
+  private static final String ANNOTATION_VALUE_JOIN_SQL =
+      " LEFT OUTER JOIN cohort_annotation_enum_value ae%d "
+      + "ON ae%d.cohort_annotation_enum_value_id = a%d.cohort_annotation_enum_value_id";
+
+  private static final String WHERE_SQL =
+      " WHERE pcs.cohort_review_id = %d";
+
+  private static final ImmutableSet<CohortStatus> REVIEWED_STATUSES =
+      ImmutableSet.of(CohortStatus.INCLUDED, CohortStatus.EXCLUDED, CohortStatus.NEEDS_FURTHER_REVIEW);
+
+  private Map<String, CohortAnnotationDefinition> getAnnotationDefinitions(CohortReview cohortReview) {
+    return Maps.uniqueIndex(
+            cohortAnnotationDefinitionDao.findByCohortId(cohortReview.getCohortId()),
+            CohortAnnotationDefinition::getColumnName);
+  }
+
+  private String getSelectAndFromSql(CohortReview cohortReview, List<String> columns,
+      Map<String, CohortAnnotationDefinition> annotationDefinitions, Map<String, String> columnAliasMap) {
+    StringBuilder selectBuilder = new StringBuilder("SELECT ");
+    StringBuilder fromBuilder = new StringBuilder("\nFROM participant_cohort_status pcs");
+    int annotationCount = 0;
+    boolean firstColumn = true;
+    for (String column : columns) {
+      if (firstColumn) {
+        firstColumn = false;
+      } else {
+        selectBuilder.append(", ");
+      }
+      if (column.equals(PERSON_ID_COLUMN)) {
+        selectBuilder.append("pcs.participant_id person_id");
+      } else if (column.equals(REVIEW_STATUS_COLUMN)) {
+        selectBuilder.append("pcs.status review_status");
+      } else {
+        if (annotationDefinitions == null) {
+          annotationDefinitions = getAnnotationDefinitions(cohortReview);
+        }
+        CohortAnnotationDefinition definition = annotationDefinitions.get(column);
+        if (definition == null) {
+          throw new BadRequestException("Invalid annotation name: " + column);
+        }
+        annotationCount++;
+        fromBuilder.append(
+            String.format(ANNOTATION_JOIN_SQL, annotationCount, annotationCount, annotationCount,
+                definition.getCohortAnnotationDefinitionId(), cohortReview.getCohortReviewId()));
+        String sourceColumn;
+        if (definition.getAnnotationType().equals(AnnotationType.ENUM)) {
+          sourceColumn = String.format("ae%d.name", annotationCount);
+          fromBuilder.append(
+              String.format(ANNOTATION_VALUE_JOIN_SQL, annotationCount, annotationCount, annotationCount));
+        } else {
+
+          String columnName = ANNOTATION_COLUMN_MAP.get(definition.getAnnotationType());
+          if (columnName == null) {
+            throw new BadRequestException("Invalid annotation type: " + definition.getAnnotationType());
+          }
+          sourceColumn = String.format("a%d.%s", annotationCount, columnName);
```

Comment:
columnName may be susceptible to SQL injection, for example. I wouldn't depend on it being validated from here (see above comment on safe SQL)

Reply:
columnName comes from ANNOTATION_COLUMN_MAP, which has values which are just the names of columns in the participant_cohort_annotation table. So I don't think there's a risk of SQL injection here.